src/utils/timer.ts

```diff
@@ -0,0 +1,71 @@
+import { logWarn } from '.';
+
+interface TimerProps {
+    /** Set a upper threshold for the timer, which will make the timer autostop. Default is 2000 milliseconds. */
+    maxTime?: number;
+    /** Optional callback. Check the this reference binding as it will default to {Timer} */
+    callback?: () => void;
+}
+
+type TimerStatus = 'Started' | 'Stopped' | 'Over time' | 'Not started';
+
+export class Timer {
+    /** Time elapsed is stored as milliseconds. */
+    private _timeElapsed = 0;
+    private _startTime?: Date;
+
+    /** An optional callback when the timer stops. */
+    private _callback?: () => void;
+
+    /**
+     * We use a max time for the timer to avoid it counting "forever".
+     * This value is configurable during construction.
+     */
+    private _maxTime: number;
+    private _timerId?: NodeJS.Timeout;
+    private _status: TimerStatus;
+
+    constructor(props?: TimerProps) {
+        if (props && props.maxTime && typeof props.maxTime === 'number' && !isNaN(props.maxTime)) {
+            this._maxTime = props.maxTime;
+        } else this._maxTime = 2000;
+
+        this._status = 'Not started';
+    }
+
+    public get timeElapsed() {
+        return this._timeElapsed;
```

Comment:
It should return current elapsed time if it's running, seems like it only has this time after it's stopped?

Reply:
Good catch, I have resolved this